src/conversions/chrono.rs

```diff
@@ -0,0 +1,573 @@
+        let seconds_offset = self.local_minus_utc();
+        let td =
+            PyDelta::new(py, 0, seconds_offset, 0, true).expect("Failed to contruct timedelta");
+        let offset = dt_timezone
+            .call1((td,))
+            .expect("Failed to call timezone with timedelta");
+        offset.into()
+    }
+}
+
+impl IntoPy<PyObject> for FixedOffset {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for FixedOffset {
+    /// Convert python tzinfo to rust [`FixedOffset`].
+    ///
+    /// Note that the conversion will result in precision lost in microseconds as chrono offset
+    /// does not supports microseconds.
+    fn extract(ob: &PyAny) -> PyResult<FixedOffset> {
+        let py_tzinfo = <PyTzInfo as PyTryFrom>::try_from(ob)?;
+        let py_timedelta = py_tzinfo.call_method1("utcoffset", (ob.py().None(),))?;
+        let py_timedelta = <PyDelta as PyTryFrom>::try_from(py_timedelta)?;
+        Ok(FixedOffset::east(py_timedelta.get_seconds()))
+    }
+}
+
+impl ToPyObject for Utc {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        timezone_utc(py).to_object(py)
+    }
+}
+
+impl IntoPy<PyObject> for Utc {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for Utc {
+    fn extract(ob: &PyAny) -> PyResult<Utc> {
+        let py_tzinfo = <PyTzInfo as PyTryFrom>::try_from(ob)?;
+        let py_utc = timezone_utc(ob.py());
+        if py_tzinfo.eq(py_utc)? {
+            Ok(Utc)
+        } else {
+            Err(PyTypeError::new_err("Not datetime.timezone.utc"))
+        }
+    }
+}
+
+#[cfg(test)]
+mod test_chrono {
+    use crate::types::*;
+    use crate::{Python, ToPyObject};
+    use chrono::offset::{FixedOffset, Utc};
+    use chrono::{DateTime, Duration, NaiveDate, NaiveTime};
+    use std::cmp::Ordering;
+
+    #[test]
+    fn test_pyo3_timedelta_topyobject() {
+        use std::panic;
+
+        Python::with_gil(|py| {
+            let check = |s, ns, py_d, py_s, py_ms| {
+                let delta = Duration::seconds(s) + Duration::nanoseconds(ns);
+                let delta = delta.to_object(py);
+                let delta: &PyDelta = delta.extract(py).unwrap();
+                let py_delta = PyDelta::new(py, py_d, py_s, py_ms, true).unwrap();
+                assert!(delta.eq(py_delta).unwrap());
+            };
+            let check_panic = |duration: Duration| {
+                assert!(panic::catch_unwind(|| {
+                    duration.to_object(py);
+                })
+                .is_err())
+            };
+
+            check(-86399999913600, 0, -999999999, 0, 0); // min
+            check(86399999999999, 999999000, 999999999, 86399, 999999); // max
+
+            check_panic(Duration::min_value());
+            check_panic(Duration::max_value());
+            // TODO: check timedelta underflow
```

Comment:
@Psykopear Thanks for taking over the PR, can you please help to take a look at this?

Reply:
Yes, I'll try to add tests for the underflow too

Reply:
So, when you say to check for timedelta (I assume it's `Duration` in this case) underflow, you mean something like this?
```rust
let delta = Duration::seconds(-(i64::MIN + 1)) - Duration::seconds(2);
```
This panics even before trying to convert it to python, but that's not something related to the pyo3-chrono integration, it's more of a `chrono` problem because we can use `std::ops::Sub` rather than `Duration::checked_sub`, which returns an `Option<Duration>`, so I don't think we need to test it here.